### PR TITLE
1st version of Engineering Practice blog

### PR DESCRIPTION
**Pull request by MerlinDarkstar**

### CODE REVIEW

**Comment by wpschaub** [comment]
@MerlinDarkstar initial review:

1. "...through the use of Sonar Cube " could be "...using SonarQube".
2. Not above, I think you mean SonarQube, not Sonar Cube :)
3. Can you remove the non *.md and *.png/jpg files from the Pull Request? I will add a gitignore file to the repo.
4. The section "# TDD and PowerShell Dojo Planning" is internal content - not suited for "public" post.

**Comment by wpschaub** [comment]
@MerlinDarkstar I am still seeing the .vs folder and all its files in the PR.

Please add PR reviewers when you are ready for us to review :)

**Review by danielrbroderick — DISMISSED** [review]

File: content/EngineeringPractice-1.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+Title: Engineering Practice Nuggets
+Date: 2023-01-10
+Category: Posts 
+Tags: agile, architecture, code, code-quality, design, engineering, technical-excellence, tdd, tips, learning, innovation
+Slug: EngineeringPractice-1
+Author: Martin M. Lacey
+Summary: Nuggets of Knowledge and Best Practices in Software Engineering
+
+When we design, build, and deploy software we aim to apply patterns and learned best practices in order to produce
+consistent and rohbust solutions that are of the highest quality and lowest possible technical dept (burden to maintain).
```
Comment (wpschaub) [inline content/EngineeringPractice-1.md]: Typo - robust

File: content/EngineeringPractice-1.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+
+---
+# Metrics and Quantitative Measures
+We have begun the process of calculating various metrics on software created at
+WSBC, using SonarQube in the build and deploy pipelines.
+These will soon be augmented by Stryker to generate mutant test cases and validate them, 
+as well as the Roslyn Analyzer for further code analysis.  
+
+These measurements will be calculated when the pipeline for a product is
```
Comment (wpschaub) [inline content/EngineeringPractice-1.md]: How about changing "will be" to "... measurements are ..."?

File: content/EngineeringPractice-1.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+These powerful nuggets of knowledge, patterns and practices, and key areas that we are evolving and 
+improving will be the subject matter explored with the goal to invigorate 
+vibrant collaboration and innovation, and indeed experimentation and tenacity 
+when things don't go as desired.
+
+
+---
+# Metrics and Quantitative Measures
+We have begun the process of calculating various metrics on software created at
```
Comment (wpschaub) [inline content/EngineeringPractice-1.md:26]: You could mention the four DevOps metrics, which are standard across DevOps ecosystem.

File: content/EngineeringPractice-1.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+---
+# Engineering Capability Maturity
+We are now developing a Capability Maturity Model for Software Engineering Practices 
+at WSBC, similar to the Maturity Model for Agile practices combined with CMM for 
+Software - and supported by Scaled Agile Framework for the Enterprise (SAFe).  
+
+We are working to identify the capabilities at each level and how to measure them, 
+and how to put processes in place or refine existing processes  to support their 
+improvement and cultivate an open culture of experiment-fail-learn and share.
```
Comment (wpschaub) [inline content/EngineeringPractice-1.md]: ... and share, as discussed in [Fail, Learn, Reset!](/fail-learn-reset.html).

File: content/EngineeringPractice-1.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+product and purpose and current version.  Clicking on the tile opens a modal 
+dialog describing the product in full, with links to download either from 
+ServiceNow or the product own download page.
+
+We are busily creating Phase II, enabling a Search capability and populating the
+table cells based on a query to live *editable* data in SharePoint.
+
+If you'd like to see what it looks like now, we have a prototype that demonstrates
+the basic User Experience - but lacking this last Phase II piece.
+Check it out here: [EDO Ce Recommended Products Catalog.](https://wsdvuxstatic.blob.core.windows.net/demos/EEOPRODUCTCATALOGUE/index.html)
```
Comment (wpschaub) [inline content/EngineeringPractice-1.md]: I would remove "If you'd like ... .html) paragraph and rather "tease" the reader with "we will explore the product catalog in a future discussion". Remember the blog posts are read by non-WSBC users and "EDO" makes no sense to them.

File: content/EngineeringPractice-1.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+Phase III - Q1/Q2 Will tie this in with the Product Radar in the Technology
```
Comment (wpschaub) [inline content/EngineeringPractice-1.md]: Again, the community reader has no context on TAP and DRG.  Be careful not to bleed internal programs in these posts, other than those we agreed to share, such as our pipelines.

**Review by wpschaub — CHANGES_REQUESTED** [review]
@MerlinDarkstar make sure to focus these blogs on the non-WSBC community readers.

**Review by wpschaub — APPROVED** [review]

**Comment by wpschaub** [comment]
@MerlinDarkstar now that I know you are using VSE I understand all the noise that is checked in. Once merged, I will remove the .vs folder alltogether.

**Review by danielrbroderick — APPROVED** [review]

**Pull request merged**